naps2: 7.4.3 -> 7.5.3 and use non-EOL dotnet

elfenermarcell opened the pull request:
This package now uses .NET 8 instead of the EOL version 6

also added wrapGAppsHook3, this should resolve #339293


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

elfenermarcell commented:
https://github.com/NixOS/nixpkgs/issues/326335

elfenermarcell commented:
Also, someone should close https://github.com/NixOS/nixpkgs/issues/249522 (It seems it wasn't closed when NAPS2 was originally packaged)

gepbird commented on pkgs/by-name/na/naps2/package.nix:
```diff
@@ -26,20 +27,8 @@ buildDotnetModule rec {
 
   executables = [ "naps2" ];
 
-  dotnet-sdk =
-    with dotnetCorePackages;
-    sdk_8_0
-    // {
-      inherit
-        (combinePackages [
-          sdk_8_0
-          sdk_6_0-bin
-        ])
-        packages
-        targetPackages
-        ;
-    };
-  dotnet-runtime = dotnetCorePackages.runtime_8_0;
```
It could be safer to specify the dotnet version to avoid breakages when the next version comes out and will be the default:
```nix
  dotnet-sdk = dotnetCorePackages.sdk_8_0;
  dotnet-runtime = dotnetCorePackages.runtime_8_0;
```

gepbird commented on pkgs/by-name/na/naps2/package.nix:
```diff
@@ -1,23 +1,24 @@
     owner = "cyanfish";
     repo = "naps2";
-    rev = "v${version}";
+    rev = "v${finalAttrs.version}";
```
Nit:
```suggestion
    tag = "v${finalAttrs.version}";
```

gepbird reviewed (changes requested):
Thanks for updating the package and using a supported dotnet version!

gepbird reviewed (approved):
Changes LGTM, tested the binary

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 378332`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>naps2</li>
  </ul>
</details>

GGG-KILLER commented on pkgs/by-name/na/naps2/package.nix:
```diff
@@ -1,23 +1,24 @@
     owner = "cyanfish";
     repo = "naps2";
-    rev = "v${version}";
+    tag = "v${finalAttrs.version}";
```
This is an incorrect usage of `finalAttrs`, as it might induce people into thinking they only need to change the `version` instead of both the `version` and `hash` (see: #310373)

elfenermarcell replied:
So, in this case what would you recommend? Just go back to `rec {}`?

GGG-KILLER replied:
Yes

GGG-KILLER reviewed (changes requested):
Everything looks good, only a minor nit.

Thanks for doing this, I opened the original issue but ended up not having time to update the package myself 😔

GGG-KILLER reviewed (approved):
~~LGTM! Thanks for this!~~

Sorry, ignore this, noticed something else.

GGG-KILLER commented on pkgs/by-name/na/naps2/package.nix:
```diff
@@ -1,23 +1,24 @@
     repo = "naps2";
-    rev = "v${version}";
+    tag = "v${version}";
     hash = "sha256-/qSfxGHcCSoNp516LFYWgEL4csf8EKgtSffBt1C02uE=";
```
Actually, shouldn't the hash be different here?

elfenermarcell replied:
Oops, accidentally did that thing from #310373
> It will be a silent incorrect behavior when the FOD is cached locally (reusing the old one)

GGG-KILLER reviewed (approved):
Everything looks good now, it still builds although I did not test the binary.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 378332`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>naps2</li>
  </ul>
</details>

fugidev commented on pkgs/by-name/na/naps2/package.nix:
```diff
@@ -67,7 +59,6 @@ buildDotnetModule rec {
     platforms = lib.platforms.linux;
     mainProgram = "naps2";
-    broken = stdenv.hostPlatform.isAarch64;  # Google.Protobuf.Tools dependency fails to build.
+    broken = stdenv.hostPlatform.isAarch64; # Google.Protobuf.Tools dependency fails to build.
```
```suggestion

```

fugidev reviewed (changes requested):
building on aarch64 is fixed by this PR 🎉

gepbird reviewed (approved):
Tested the app again and looks good
(the check is only failing due to a GH bug)
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 378332`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>naps2</li>
  </ul>
</details>

elfenermarcell commented:
It seems upstream released new versions and they fail to build using `dotnet-sdk = dotnetCorePackages.sdk_8_0; dotnet-runtime = dotnetCorePackages.runtime_8_0;`

Unfortunately I'm having internet issues so it might be a bit of time before I can look at this.

elfenermarcell committed:
naps2: 7.4.3 -> 7.5.3 and use non-EOL dotnet

use .NET 8 instead of the EOL version 6

added wrapGAppsHook3, this should resolve #339293

no longer broken on aarch64

elfenermarcell commented:
Resolved the merge conflict. This is still version 7.5.3, the last version of naps2 before version 8. Looks like updating to the latest would require some combination of .NET versions, but I think this is fine for now; at least it allows installing naps2 from nixpkgs without using EOL stuff.

gepbird commented:
> Looks like updating to the latest would require some combination of .NET versions, but I think this is fine for now; at least it allows installing naps2 from nixpkgs without using EOL stuff.

I agree, it's best to only fix the EOL warning if updating to a new version is not trivial

gepbird commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 378332`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>naps2</li>
  </ul>
</details>

Friendly ping @corngood

corngood commented:
This adds .net 6.0 runtime (and app host) to deps.json, so it's not really removing the EOL dependency.  It's probably not publishing anything with that runtime, but I don't understand why it's needed:

```
Executing dotnetConfigureHook
  Determining projects to restore...
/build/source/NAPS2.Images/NAPS2.Images.csproj : error NU1102: Unable to find package Microsoft.NETCore.App.Runtime.linux-x64 with version (= 6.0.36) [/build/source/NAPS2.App.Gtk/NAPS2.App.Gtk.csproj]
```

I would have thought it would only need the ref package.

gepbird reviewed (approved):
Thanks to both of you!

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 378332`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>naps2</li>
  </ul>
</details>

I may look into building the latest version in a week

fugidev reviewed (approved):
works on aarch64 too :+1:

Thatoo commented:
@thiagokokada can you cherry-pick in order to backport to 24.11?
